dcos_installer/backend.py

```diff
@@ -43,6 +43,28 @@ def do_configure(config_path=CONFIG_PATH):
     return 0
 
 
+def generate_node_upgrade_script(current_version, config_path=CONFIG_PATH):
+
+    if current_version is None:
+        log.error('Must provide the current version of the cluster')
+        return 0
+
+    config = Config(config_path)
+
+    validate_gen = config.do_validate(include_ssh=False)
+    if len(validate_gen) > 0:
+        for key, error in validate_gen.items():
+            log.error('{}: {}'.format(key, error))
+        return 1
+
+    gen_out = config_util.do_configure(config)
```

Comment:
gen_out helps not regenerate the same things over again to reference in upgrades. this function is basically do_configure.